webapp-django/crashstats/manage/forms.py

```diff
@@ -40,6 +42,84 @@ class FilterEventsForm(BaseForm):
     action = forms.CharField(required=False)
 
 
+class FilterAPITokensForm(BaseForm):
+
+    user = forms.CharField(required=False)
+    key = forms.CharField(required=False)
+    expired = forms.CharField(required=False)
+
+    def clean_expired(self):
+        value = self.cleaned_data['expired']
+        if value not in ('yes', 'no', ''):
+            raise forms.ValidationError('Invalid value')
+        return value
+
+
+class APITokenForm(BaseModelForm):
+
+    user = forms.CharField(required=True)
+    expires = forms.ChoiceField(required=True)
+
+    class Meta:
+        model = Token
+        fields = ('user', 'expires', 'notes', 'permissions')
+
+    def __init__(self, *args, **kwargs):
+        self.possible_permissions = kwargs.pop('possible_permissions', [])
+        expires_choices = kwargs.pop('expires_choices', [])
+        super(APITokenForm, self).__init__(*args, **kwargs)
+        self.fields['permissions'].choices = (
+            (x.pk, x.name) for x in self.possible_permissions
+        )
+        self.fields['user'].widget = forms.widgets.TextInput()
+        self.fields['user'].label = 'User (by email)'
+        self.fields['expires'].choices = expires_choices
+
+    def clean_user(self):
+        value = self.cleaned_data['user']
+        try:
+            user = User.objects.get(email__istartswith=value.strip())
```

Comment:
You should be worried. :) 
Sort of. 
My excuse, if you can call it that, was I that I forgot to address it. I don't think I want(ed) change the code but I should have responded. 

The example you mention is extremely rare. That you have users with `@mail.co` and `@mail.com` on the same project is super rare. 

Let's address this if and when it becomes a problem. 

Also, it wouldn't have been a problem if select2 didn't have a bug that prevented me from writing a select2 widget that uses AJAX and can accept 1 value into a type input field. It's a [known open bug](https://github.com/select2/select2/issues/2800).